### PR TITLE
feat(file): Add UFolderButton and UImageButton in the ui library and showroom

### PR DESCRIPTION
**Pull request by hyj1204**
## Description
1. Add `UFolderButton` widget in the ui library and showroom
It is used to show a folder in file session. It is always a folder graph shows in the center.
When the file number is less than 2, the text show 'item' instead of 'items'.
The two icons on the top is to show the folder if it is linked, favored. (Linked statues may not exist in the future, but I leave it on until future change)
The bottom right icons only show when the folder is in the 'delete' status. It switch is statues by taping. 
Currently, this icon is only for showroom, It may change when it implement in the app.
 
image

https://user-images.githubusercontent.com/14248245/180564901-45461e07-ea4b-40f8-9f27-b02ada6e0514.mov

3. Add `UImageButton` widget in the ui library and showroom
Similar with UFolderButton, UImageButton shows an image preview instead a folder graph.

https://user-images.githubusercontent.com/14248245/180565651-87af007e-e395-4f8c-a706-3dd1de53a305.mov

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/ui_library/lib/widgets/buttons/u_folder_button/u_folder_button.dart [hunk 1]
```diff
@@ -0,0 +1,139 @@
+import 'package:flutter/material.dart';
+import 'package:ui_library/ui_library_export.dart';
+
+class UFolderButton extends StatefulWidget {
+  ///Used in file session to represent a folder
+  ///
+  ///When [isDeleting] set to ture, it will show a checkbox
+  ///
+  ///You can use [isSelected] in UFolderButtonState to get it is checked or not
+  ///
```
Comment (stavares843) [inline packages/ui_library/lib/widgets/buttons/u_folder_button/u_folder_button.dart:10]: do we need the   /// if the line is empty?

File: packages/ui_library/lib/widgets/buttons/u_image_button/u_image_button.dart [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
+import 'package:ui_library/ui_library_export.dart';
+
+class UImageButton extends StatefulWidget {
+  ///Used in file session to preview image
+  ///
```
Comment (stavares843) [inline packages/ui_library/lib/widgets/buttons/u_image_button/u_image_button.dart:6]: do we need the /// if the line is empty?

**Review by phillsatellite — APPROVED** [review]

File: packages/ui_library/lib/widgets/buttons/u_folder_button/u_folder_button.dart (hunk 1 above)
Comment (hyj1204) [reply packages/ui_library/lib/widgets/buttons/u_folder_button/u_folder_button.dart:10]: '///'  here for a new line. If I don't add it, the in-code doc will look like this
image
so I added it to separate the sentence, then it looks like here right now
image

File: packages/ui_library/lib/widgets/buttons/u_image_button/u_image_button.dart (hunk 2 above)
Comment (hyj1204) [reply packages/ui_library/lib/widgets/buttons/u_image_button/u_image_button.dart:6]: same as previous one

File: packages/ui_library/lib/widgets/buttons/u_folder_button/u_folder_button.dart (hunk 1 above)
Comment (stavares843) [reply packages/ui_library/lib/widgets/buttons/u_folder_button/u_folder_button.dart:10]: thanks for explaining 🔨 

doesn't flutter support multi line comment?

such as 

/*

*/

Comment (hyj1204) [reply packages/ui_library/lib/widgets/buttons/u_folder_button/u_folder_button.dart:10]: I just check the documents in Flutter. Here is the link
https://dart.dev/guides/language/effective-dart/documentation

They do have` /** ... */ ` for multiple line DOC comments, but they still suggest to use `///` for serval reasons. 
When we use the vscode, it auto generate ///  when you tap 'return' in DOC comments, so I feel it is easier to us to use it.
image

**Review by lgmarchi — APPROVED** [review]
Very good. Approved. 
image
image

**Pull request merged**